mirai-core-mock/test/RemoteFileV2Test.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2019-2022 Mamoe Technologies and contributors.
+ *
+ * 此源代码的使用受 GNU AFFERO GENERAL PUBLIC LICENSE version 3 许可证的约束, 可以在以下链接找到该许可证.
+ * Use of this source code is governed by the GNU AGPLv3 license that can be found through the following link.
+ *
+ * https://github.com/mamoe/mirai/blob/dev/LICENSE
+ */
+
+package net.mamoe.mirai.mock.test
+
+import com.google.common.jimfs.Configuration
+import com.google.common.jimfs.Jimfs
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.toList
+import net.mamoe.mirai.contact.MemberPermission
+import net.mamoe.mirai.event.events.GroupMessageEvent
+import net.mamoe.mirai.message.data.FileMessage
+import net.mamoe.mirai.mock.internal.remotefile.v2.MockRemoteFiles
+import net.mamoe.mirai.mock.internal.txfs.TxFileSystemImpl
+import net.mamoe.mirai.mock.utils.simpleMemberInfo
+import net.mamoe.mirai.utils.ExternalResource.Companion.toExternalResource
+import net.mamoe.mirai.utils.cast
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Test
+import java.nio.file.FileSystem
+import kotlin.test.assertEquals
+
+internal class RemoteFileV2Test : MockBotTestBase() {
+    private val tmpfs: FileSystem = Jimfs.newFileSystem(Configuration.unix())
+    private val disk = bot.tmpFsServer.fsDisk
+    private val g = bot.addGroup(11L, "a")
+    private val fsys = TxFileSystemImpl(disk.cast())
+    private val files = MockRemoteFiles(g, fsys)
+
+    @AfterEach
+    internal fun release() {
+        tmpfs.close()
+    }
+
+    @Test
+    internal fun listFileAndFolder() = runTest {
+        val f = files.root.createFolder("test1")
+        files.root.createFolder("test2")
+        val ff = f.uploadNewFile("test.txt", "cc".toByteArray().toExternalResource().toAutoCloseable())
```

Comment:
避免需要花时间读赋值代码才能看懂的命名 (ff).
```suggestion
        val folder = files.root.createFolder("test1")
        files.root.createFolder("test2")
        val file = f.uploadNewFile("test.txt", "cc".toByteArray().toExternalResource().toAutoCloseable())
```
记得替换 reference, 不要直接 apply suggestion